### ISSUE TITLE
Timeout on Request error macOS 12.1

### ISSUE THREAD

**Issue by Dark000Soldier** [Gkiokan/ps4-remote-pkg-sender#6]
On my macOS 12.1 Server is running but i always got Timeout on Request error !
LOG:

1 file found
1 file serving
Server is running on port 192.xx.xx.xx:9999

**Comment by Dark000Soldier** [comment]
Firewall is off on my mac already and I checked on Accessibility but I did not find any request to permission on any catalog.

**Comment by Dark000Soldier** [comment]
I think the problem on my side is something else. I have another Payload Sender and it work on iOS and Mac, the version runs on iOS no problem, but the one on mac doesn’t!
Even using VT win11 doesn’t help, because when run an app on VT win11 on mac, I got same problem.
I think my mac for whatever reason doesn’t allow this apps to search and access other apps on network. So the app can’t find PS4.
I could download the file.pkg from my the server your app creat with no problem. Which means the server work correctly, the only problem that my mac block the access to ps4.
I wonder if there an terminal commend to give your app permission to access network.

**Comment by Dark000Soldier** [comment]
> Try restarting the RPI on PS4. Fixed the issue on my side. Timeout could be also a general error not finding the ps4 or the RPI is not accessible (e.g. RPI/PS4 change to standby)
> 
> Next update will have a custom config value to configure the timeout individual.

What is RPI ?

**Comment by Gkiokan** [comment]
RPI = Remote Package Installer (the Homebrew app from flatZ)

Can you check your Security and Privacy settings if there is something blocking the network connection?
You could try to ping your PS4 IP Adress from terminal
```
 ❯ ping 192.168.43.174                                                                                             [15:53:23]
PING 192.168.43.174 (192.168.43.174): 56 data bytes
64 bytes from 192.168.43.174: icmp_seq=0 ttl=64 time=189.763 ms
64 bytes from 192.168.43.174: icmp_seq=1 ttl=64 time=47.763 ms
64 bytes from 192.168.43.174: icmp_seq=2 ttl=64 time=7.793 ms
```

**Comment by Dark000Soldier** [comment]
Okay yes I got exactly as you send even time is under 1ms.

**Comment by Gkiokan** [comment]
So your mac can access the PS4 but the apps doesn't.

Mac has some options in the system preferences to block connections but thoose seems to be incomming and not outgoing connections. You can check them on your Firewall Extended Options.

![image](https://user-images.githubusercontent.com/7249224/148079737-e1cbe0d8-54b4-45b0-896c-fc29087eb6b0.png)

**Comment by Dark000Soldier** [comment]
My firewall is off, shouldn’t that mean that all connection are allowed? Or better to enable it and then app by app allowing them?

**Comment by Gkiokan** [comment]
Yea you are right. But if it's not the the Firewall, what else could be block the connection then. 

There are netstat commands but as I know you can only check running services and ports on your local machine with it.

Mac has also that "unknown developer" dialog message when you open downloaded apps on first run.
Something like "Do you want to allow that app to have access to this and that." Maybe it's blocked by it?

**Comment by Dark000Soldier** [comment]
Now I got error 2157510677

**Comment by Gkiokan** [comment]
2157510677 means already installed.

There are 3 error codes known for now and they are not documented. I've patched them in the last version. 

Just refining some stuff and will release that too.

```
      if(code==2157510681)
        message = code + " | task doesn't exist (?)"

      if(code==2157510663)
        message = code + " | already installed (?)"

      if(code==2157510677)
        message = code + " | It seems to be installed already."
```

**Comment by Dark000Soldier** [comment]
Neither I, I had this problem with VT win too, running some Power-shell commends give me network errors too before. It is M1 chip, maybe something with Apple Silicone security thing. I don’t know

**Comment by Gkiokan** [comment]
Nice. Well, I'm glad it worked out for you. Feel free to open a new Issue when something happens.
